### PR TITLE
update 02-client proposal_test

### PR DESCRIPTION
**Pull request by gnehcein**

### CODE REVIEW

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 1]
```diff
@@ -1,85 +1,231 @@
...
+		types.NewHeight(0, uint64(suite.chainA.CurrentHeader.Height-1)),
+		suite.chainA.CurrentHeader.Time, suite.chainA.Vals,
+		suite.chainA.Vals, suite.chainA.Signers,
+	)
+	testCases := []struct {
+		msg      string
+		malleate func()
+	}{
+		{
+			"success,create new client",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:23]:
```suggestion
			"success, create new client",
```

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 2]
```diff
@@ -1,85 +1,231 @@
...
+				var proposal *types.CreateClientProposal
+				var err error
+				proposal, err = types.NewCreateClientProposal("test", "test", "test", clientState, consensusState)
+				suite.NoError(err)
+				err = suite.chainA.App.TIBCKeeper.ClientKeeper.HandleCreateClientProposal(suite.chainA.GetContext(), proposal)
+				suite.NoError(err)
+			},
+		},
+		{
+			"fail,A client for this chainname already exists",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:38]: ditto.

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 3]
```diff
@@ -1,85 +1,231 @@
...
+				proposal, err = types.NewCreateClientProposal("test", "test", "test", clientState, consensusState)
+				suite.NoError(err)
+				err = suite.chainA.App.TIBCKeeper.ClientKeeper.HandleCreateClientProposal(suite.chainA.GetContext(), proposal)
+				suite.NoError(err)
+				err = suite.chainA.App.TIBCKeeper.ClientKeeper.HandleCreateClientProposal(suite.chainA.GetContext(), proposal)
+				suite.Error(err)
+			},
+		},
+		{
+			"success,get client and compare",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:55]: ditto.

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 4]
```diff
@@ -1,85 +1,231 @@
...
+				var proposal *types.UpgradeClientProposal
+				var err error
+				proposal, err = types.NewUpgradeClientProposal("test", "test", "test", clientState, consensusState)
+				suite.NoError(err)
+				err = suite.chainA.App.TIBCKeeper.ClientKeeper.HandleUpgradeClientProposal(suite.chainA.GetContext(), proposal)
+				suite.Error(err)
+			},
+		},
+		{
+			"success,client and consensus are existing",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:111]: ditto.

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 5]
```diff
@@ -1,85 +1,231 @@
...
+		types.NewHeight(0, uint64(suite.chainA.CurrentHeader.Height-1)),
+		suite.chainA.CurrentHeader.Time, suite.chainA.Vals,
+		suite.chainA.Vals, suite.chainA.Signers,
+	)
+	testCases := []struct {
+		msg      string
+		malleate func()
+	}{
+		{
+			"success,exist client",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:165]: ditto.

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 6]
```diff
@@ -1,85 +1,231 @@
...
-	// 		}
-	// 	})
-	// }
+				// get relayers and compare
+				relayers2 := suite.chainA.App.TIBCKeeper.ClientKeeper.GetRelayers(suite.chainA.GetContext(), "test")
+				suite.Equal(relayers, relayers2)
+			},
+		},
+		{
+			"fail,no client",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:192]: ditto.

File: modules/tibc/core/02-client/keeper/proposal_test.go [hunk 7]
```diff
@@ -1,85 +1,231 @@
...
+			func() {
+				relayers := []string{"xxx", "yyy"}
+				relayerProposal := types.NewRegisterRelayerProposal("test", "test", "test", relayers)
+				err := suite.chainA.App.TIBCKeeper.ClientKeeper.HandleRegisterRelayerProposal(suite.chainA.GetContext(), relayerProposal)
+				suite.Error(err)
+			},
+		},
+		{
+			"fail,no-existing client",
```
Comment (chengwenxi) [inline modules/tibc/core/02-client/keeper/proposal_test.go:201]: ditto.

**Review by chengwenxi — APPROVED** [review]
LGTM, some minor typo issues.

File: modules/tibc/core/02-client/keeper/proposal_test.go (hunk 2 above)
Comment (gnehcein) [reply modules/tibc/core/02-client/keeper/proposal_test.go]: just say what‘s the error is ok

**Review by chengwenxi — APPROVED** [review]

**Pull request merged**